lib/ecto/migration.ex

```diff
@@ -801,6 +801,14 @@ defmodule Ecto.Migration do
     validate_type!(subtype)
   end
 
+  defp validate_type!({type, subtype}) when is_atom(type) and is_list(subtype) do
+    for t <- subtype, do: validate_type!(t)
+  end
+
+  defp validate_type!({type, subtype}) when is_atom(type) and is_tuple(subtype) do
```

Comment:
@josevalim so we can switch back to variant introduced in [1782](https://github.com/elixir-ecto/ecto/pull/1782). Am I right?